dag_in_context/src/utility/terms.egg

```diff
@@ -1,43 +1,78 @@
 (ruleset terms)
+(ruleset terms-helpers)
+(ruleset terms-helpers-helpers)
 
 (sort TermAndCost)
 (function Smaller (TermAndCost TermAndCost) TermAndCost)
 
 (function ExtractedExpr (Expr) TermAndCost
   :merge (Smaller old new))
+;; potential extractions- use so that we don't modify the
+;; timestamp of ExtractedExpr when the costs are equal
```

Comment:
what timestamp? can you explain this a little more?

Reply:
Added comments! This is egglog's internal timestamp of when this row was changed in the table. Changing it means we'll match on it again.